pkg/plugins/plugins.go

```diff
@@ -204,7 +204,11 @@ func (p *Plugin) implements(kind string) bool {
 func loadWithRetry(name string, retry bool) (*Plugin, error) {
 	registry := NewLocalRegistry()
 	start := time.Now()
-
+	var testTimeOut int
+	if name == testNonExistingPlugin {
+		// override the timeout in tests
+		testTimeOut = 2
+	}
```

Comment:
I'm not a big fan of the "test special cases" in the "real" code, but I don't think I have an easy alternative that's also parallel-friendly, so I'll just leave it as a grumpy non-blocking comment 😄

Reply:
I agree with your 100%. It's been a while since I wrote this originally, but I recall that I went back-and-forth a few times to find alternatives, but I wanted to avoid a package-level variable in production code as that felt more risky (besides the parallel-friendly point).